### PR TITLE
[TF] Use decl names in graph node names for better debugging support

### PR DESCRIPTION
**Pull request by rxwei**
Originally names of TF graph nodes generated by TFLowerGraph were mangled SIL function names. Now we change those to the AST names with argument labels, with a "stack trace" of inlining separated by `/`. This makes TensorBoard produce a hierarchical view of the graph for better debugging and visualization.

Known problem: `SILDebugScope::InlinedCallsite` is always `nullptr` even though we are using the performance inliner. This PR has not changed that behavior so we'll investigate it later.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

**Review by mhong — APPROVED** [review]
Nice!

File: lib/SILOptimizer/Mandatory/TFLowerGraph.cpp [hunk 1]
```diff
@@ -596,7 +611,20 @@ std::string TFGraphLowering::getUniqueName(SILDebugLocation loc,
       if (fnName.endswith(".device_partition"))
         fnName = fnName.drop_back(strlen(".device_partition"));
 
-      name += "." + fnName.str() + "." + llvm::utostr(lineCol.first);
+      // Separate functions using '/' so that TensorBoard can treat it as a
+      // hierarchical separator.
+      name += '/';
```
[the remark this answers is not in the document]
Comment (rxwei) [reply lib/SILOptimizer/Mandatory/TFLowerGraph.cpp]: Done.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

**Review by lattner — APPROVED** [review]
This LGTM.  It looks like the inline locations are working in some cases, assuming this test passes: "input: "op/main/baz/bar.".  If this is not working, then I'd suggest stepping through your loop and seeing where it fails.  It could be that we're getting an early break, and the loop should continue instead of breaking.

**Comment by rxwei** [comment]
@lattner Unfortunately the test did not work so I had it commented out.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow and merge

**Comment by rxwei** [comment]
@swift-ci please test tensorflow and merge

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**